main.rb

```diff
@@ -0,0 +1,87 @@
+require_relative "word.rb"
+require_relative "explosive.rb"
+
+class User
+  attr_reader :letter, :correct_guesses, :dashes, :guesses
+
+  # initialize instances with word and image
+  # calling word and image class
+  # necessary setup for game function
+  def initialize(word, image)
+    @word = word
+    @correct_guesses = Array.new(word.game_word.length)
+    @dashes = Array.new(word.game_word.length, "-")
+    @guesses = []
+    @counter = 0
+    @image = image
+  end
+
+  def user_input
+    #loop to confirm user has included a new guess
+    # if new guess, added to guess array
+    loop do
```

Comment:
I like that this is its own method, but it ends up being pretty complex. Would it be possible to break this up further into several smaller methods? For example, you could have:

- A method that gets input from the user and verifies it's OK
- A method that handles a correct guess
- A method that handles an incorrect guess

It seems like you're almost on that track with the `right_guess` and `wrong_guess` methods below.